Problems compiling faust2 on linux

I'm having some problems compiling current faust2 branch on Linux.

For starters I have to add `--std=c++11` to CXXFLAGS, which I do with
`CXXFLAGS="-O3 -Wall -Wuninitialized --std=c++11" make`

Then a typo error comes up in `compiler/generator/llvm/llvm_dsp_aux.cpp`, here's my patch:
```diff --git a/compiler/generator/llvm/llvm_dsp_aux.cpp b/compiler/generator/llvm/llvm_dsp_aux
index c4e9075..b0c0914 100644
--- a/compiler/generator/llvm/llvm_dsp_aux.cpp
+++ b/compiler/generator/llvm/llvm_dsp_aux.cpp
@@ -420,7 +420,7 @@ void llvm_dsp_factory_aux::startLLVMLibrary()
 #endif
 #if (!defined(LLVM_35) && !defined(LLVM_36) && !defined(LLVM_37) && !defined(LLVM_38) && !d
         if (!llvm_start_multithreaded()) {
-            std::cerr << "llvm_start_multithreaded error...\n");
+            std::cerr << "llvm_start_multithreaded error...\n";
         }
 #endif
     }
```

Then I get 
```generator/llvm/llvm_dsp_aux.cpp: In member function ‘bool llvm_dsp_factory_aux::initJIT(std::string&)’:
generator/llvm/llvm_dsp_aux.cpp:682:82: error: operands to ?: have different types ‘std::string {aka std::basic_string<char>}’ and ‘llvm::StringRef’
         builder.setMCPU((cpu == "") ? llvm::sys::getHostCPUName() : StringRef(cpu));
                                                                                  ^
```

I'm using a fairly ancient version of Ubuntu as my build machine, and trying to compile with gcc. That would be the problem, I guess.

Is the master branch functionally equivalent to faust2, in respect of supported language features? Will faust2 become the new master branch at some point?

I am happy to either stick to master/faust1, or switch to faust2 and build on a more modern machine, unless I'm missing something and this should actually work.

Regarding the first error requiring `--stdc++11`, the error message is
```generator/Text.cpp: In function ‘std::string T(float)’:
generator/Text.cpp:183:30: error: ‘max_digits10’ is not a member of ‘std::numeric_limits<float>’
     num << std::setprecision(std::numeric_limits<float>::max_digits10) << n;
                              ^
```
The third error goes away with:
```@@ -679,7 +679,7 @@ bool llvm_dsp_factory_aux::initJIT(string& error_msg)
     #if defined(LLVM_34)
         builder.setMCPU((cpu == "") ? llvm::sys::getHostCPUName() : cpu);
     #else
-        builder.setMCPU((cpu == "") ? llvm::sys::getHostCPUName() : StringRef(cpu));
+        builder.setMCPU((cpu == "") ? llvm::sys::getHostCPUName() : static_cast<std::basic_
     #endif
```

I'll make a pull request.

--- a/compiler/generator/llvm/llvm_dsp_aux.cpp
+++ b/compiler/generator/llvm/llvm_dsp_aux.cpp
@@ -751,7 +751,7 @@ bool llvm_dsp_factory_aux::initJIT(string& error_msg)
     #if defined(LLVM_34)
         builder.setMCPU((cpu == "") ? llvm::sys::getHostCPUName() : cpu);
     #else
-        builder.setMCPU((cpu == "") ? llvm::sys::getHostCPUName() : StringRef(cpu));
+        builder.setMCPU((cpu == "") ? llvm::sys::getHostCPUName() : static_cast<std::basic_string<char>>(StringRef(cpu)));
         //std::cout << "llvm::sys::getHostCPUName()" << llvm::sys::getHostCPUName().data() << std::endl;
     #endif
         

